meson.build

```diff
@@ -535,10 +535,16 @@ endif
 
 root_incdir = include_directories('.')
 
-if get_option('gtkdoc')
-  gtkdocscan = find_program('gtkdoc-scan', required : true)
-  subdir('docs')
+if get_option('docs') == 'gtkdoc'
+  gtkdocscan = find_program('gtkdoc-scan')
+elif get_option('docs') == 'docgen'
+  gidocgen_dep = dependency('gi-docgen',
+    version: '>= 2021.1',
+    fallback: ['gi-docgen', 'dummy_dep'],
+  )
+  gidocgen = find_program('gi-docgen')
```

Comment:
I think you need a third option for anything else to error out, "Unknown doc type" or something?

Reply:
Or does meson enforce the choices?

Reply:
meson configure -Ddocs=XXXX
    ERROR: Value "XXXX" (of type "string") for combo option "developer documentation type" is not one of the choices. Possible choices are (as string): "none", "gtkdoc", "docgen".